components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java

```diff
@@ -0,0 +1,482 @@
+        HistoricProcessInstanceQuery historicProcessInstanceQuery = BPMNOSGIService.getHistoryService()
+                .createHistoricProcessInstanceQuery().processInstanceTenantId(str)
+                .processDefinitionId(pId).finished();
+        //Get the count of the complete process instances
+        long noOfHistoricInstances = historicProcessInstanceQuery.count();
+
+        //If the deployed process doesnot have any completed process instances --> Ignore
+        if (noOfHistoricInstances == 0) {
+            response.setData(taskListForProcess);
+        }
+        //If the deployed process has completed process instances --> then
+        else {
+
+            BPMNTaskInstance tInstance = new BPMNTaskInstance();
+            //Get the list of completed tasks/activities in the completed process instance by passing the
+            //process definition id of the process
+            List<HistoricTaskInstance> taskList = BPMNOSGIService.getHistoryService().
+                    createHistoricTaskInstanceQuery().taskTenantId(str).processDefinitionId(pId)
+                    .processFinished().list();
+            //Iterate through each completed task/activity and get the task name and duration
+            for (HistoricTaskInstance taskInstance : taskList) {
+                //Get the task name
+                String taskKey = taskInstance.getTaskDefinitionKey();
+                //Get the time duration taken for the task to be completed
+                long taskDuration = taskInstance.getDurationInMillis();
+
+                if (map.containsKey(taskKey)) {
+                    long tt = map.get(taskKey);
+                    map.put(taskKey, taskDuration + tt);
+                } else {
+                    map.put(taskKey, taskDuration);
+                }
+                //Iterating Task List finished
+            }
+            Iterator iterator = map.keySet().iterator();
+            while (iterator.hasNext()) {
+                String key = iterator.next().toString();
+                double value = map.get(key) / noOfHistoricInstances;
+                tInstance = new BPMNTaskInstance();
+                tInstance.setTaskDefinitionKey(key);
+                tInstance.setAverageTimeForCompletion(value);
+                taskListForProcess.add(tInstance);
+            }
+
+            response.setData(taskListForProcess);
+        }
+        return response;
+    }
+
+    /**
+     * Task variation over time i.e. tasks started and completed over the months
+     *
+     * @return array with the no. of tasks started and completed over the months
+     */
+    @GET
+    @Path("/taskVariation/")
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    public ResponseHolder taskVariationOverTime() {
+        ResponseHolder response = new ResponseHolder();
+        List list = new ArrayList();
+        String[] MONTHS = {"Jan", "Feb", "March", "April", "May", "June", "July", "Aug", "Sep", "Oct", "Nov", "Dec"};
```

Comment:
This MONTHS array should be a class level constant. Then we don't need to define it in each method.